pkg/app/impl/authz.go

```diff
@@ -267,30 +270,37 @@ func (s *AuthorizerServer) Is(ctx context.Context, req *authorizer.IsRequest) (*
 	}
 
 	results, err := query.Eval(ctx, rego.EvalInput(input))
-
 	if err != nil {
 		return resp, aerr.ErrBadQuery.Err(err).Msgf("query evaluation failed: %s", queryStmt)
-	} else if len(results) == 0 {
+	}
+	if len(results) == 0 {
 		return resp, aerr.ErrBadQuery.Err(err).Msgf("undefined results: %s", queryStmt)
 	}
 
-	v := results[0].Bindings["x"]
-	outcomes := map[string]bool{}
+	for i, d := range req.PolicyContext.Decisions {
+		v, ok := results[0].Bindings[fmt.Sprintf("x%d", i)]
+		if !ok {
+			return nil, errors.Wrapf(err, "failed getting binding for decision [%s]", d)
+		}
+
+		outcome, ok := v.(bool)
+		if !ok {
+			return nil, errors.Wrapf(err, "failed getting outcome for decision [%s]", d)
```

Comment:
Consider changing the message to indicate that the rule evaluated to an unexpected type and include the value in the log message:
```go
return nil, errors.Wrapf(err, "non-boolean outcome for decision [%s]: %s", d, v)
```

Reply:
Thanks, fixed